core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractStreamProcessingStrategyFactory.java

```diff
@@ -187,19 +186,26 @@ public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor function)
                       flowConstruct.getName(), shutdownTimeout);
           processor.forceShutdown();
         }
-        try {
-          if (!completionLatch.await(max(start - currentTimeMillis() + shutdownTimeout, 0l), MILLISECONDS)) {
-            LOGGER.warn("Subscribers of ProcessingStrategy for flow '{}' not completed in {} ms", flowConstruct.getName(),
-                        shutdownTimeout);
-          }
-        } catch (InterruptedException e) {
-          currentThread().interrupt();
-          throw new MuleRuntimeException(e);
-        }
-
+        awaitSubscribersCompletion(flowConstruct, shutdownTimeout, completionLatch, start);
       }, createOnEventConsumer(), bufferSize);
     }
 
+    protected void awaitSubscribersCompletion(FlowConstruct flowConstruct, final long shutdownTimeout,
+                                              CountDownLatch completionLatch, long startMillis) {
+      try {
+        if (!completionLatch.await(max(startMillis - currentTimeMillis() + shutdownTimeout, 0l), MILLISECONDS)) {
+          LOGGER.warn("Subscribers of ProcessingStrategy for flow '{}' not completed in {} ms", flowConstruct.getName(),
+                      shutdownTimeout);
+        }
+      } catch (InterruptedException e) {
+        LOGGER.warn("Subscribers of ProcessingStrategy for flow '{}' not completed before thread interruption",
+                    flowConstruct.getName(),
+                    shutdownTimeout);
+        currentThread().interrupt();
+        return;
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this redundant jump. ([squid:S3626](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS3626))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7706&issues=AWkMuk_RotIXnmsU4kyO&open=AWkMuk_RotIXnmsU4kyO)

[ISSUE_KEY]: AWkMuk_RotIXnmsU4kyO